tasks/consensus/task.go

```diff
@@ -0,0 +1,57 @@
+package consensus
+
+import (
+	"context"
+
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/sentinel-visor/model"
+	"github.com/filecoin-project/sentinel-visor/model/consensus"
+	visormodel "github.com/filecoin-project/sentinel-visor/model/visor"
+	"go.opentelemetry.io/otel/api/global"
+	"go.opentelemetry.io/otel/label"
+)
+
+type Task struct {
+}
+
+func NewTask() *Task {
+	return &Task{}
+}
+
+func (t *Task) ProcessTipSets(ctx context.Context, child, parent *types.TipSet) (model.Persistable, *visormodel.ProcessingReport, error) {
+	_, span := global.Tracer("").Start(ctx, "ProcessBlocks")
+	if span.IsRecording() {
+		span.SetAttributes(label.String("child", child.String()), label.Int64("height", int64(child.Height())))
+		span.SetAttributes(label.String("parent", parent.String()), label.Int64("height", int64(parent.Height())))
+	}
+	defer span.End()
+
+	var pl consensus.ChainConsensusList
+	for epoch := child.Height(); epoch > parent.Height(); epoch-- {
+		if child.Height() == epoch {
+			pl = append(pl, &consensus.ChainConsensus{
+				Height:       int64(epoch),
+				StateRoot:    child.ParentState().String(),
+				ParentTipSet: child.Parents().String(),
+				TipSet:       child.Key().String(),
+			})
+		} else {
+			// null round no tipset
+			pl = append(pl, &consensus.ChainConsensus{
+				Height:       int64(epoch),
+				StateRoot:    parent.ParentState().String(),
+				ParentTipSet: parent.Parents().String(),
+				TipSet:       "",
+			})
+		}
+	}
+	rp := &visormodel.ProcessingReport{
```

Comment:
Don't we need a report for each epoch so we know there are no gaps?

Reply:
Yup, forgot to add that. Separately, we need processing reports for tasks when they are enabled but don't execute, meaning there was no work for the task to perform. This second part will need its own PR.

Reply:
addressed in latet commit